elastic/kibana
# Introduce Kerberos authentication provider.

Introduce Kerberos authentication provider.

-------------------------

## How to test

In case it's useful, here is how I test it locally:

1. Prepare all configs/KDC/browser

```
/etc/krb5.conf
-----------------
[libdefaults]
	default_realm = TEST.ELASTIC.CO

[realms]
	TEST.ELASTIC.CO = {
		admin_server = localhost
		kdc = localhost
		default_principal_flags = +preauth
	}

[domain_realm]
	localhost = TEST.ELASTIC.CO
```

```
/etc/krb5.keytab (not actual content, just list of custom principals)
-----------------
HTTP/localhost@TEST.ELASTIC.CO
host/kerberos.test.elastic.co@TEST.ELASTIC.CO
tester@TEST.ELASTIC.CO
```

```
about:config in Firefox (note that SPNEGO doesn't work in Firefox Private Mode)
-----------------
network.negotiate-auth.trusted-uris: localhost
```

2. Create TGT with `kinit tester`

3. Run Elasticsearch with (keytab file will be automatically copied into `config` directory):
```bash
$ ES_JAVA_OPTS="-Djava.security.krb5.conf=/some-accessible-path/krb5.conf" yarn es snapshot \
    --license trial \
    -E xpack.security.authc.token.enabled=true \
    -E xpack.security.authc.realms.kerberos.kerb1.keytab.path=/some-accessible-path/krb5.keytab
```

4. Add role mapping for users authenticated with Kerberos realm, e.g.: 
```
POST http://localhost:9200/_security/role_mapping/krb5
Content-Type: application/json

{
  "roles": [ "superuser" ],
  "enabled": true,
  "rules": { "field" : { "realm.name" : "kerb1" } }
}
```

5. Run Kibana with `xpack.security.authProviders: [kerberos, basic]`

**Replaces: https://github.com/elastic/kibana/pull/27008**
**Fixes: https://github.com/elastic/kibana/issues/18188**
